src/files/extraction/helpers.ts

```diff
@@ -0,0 +1,202 @@
+    }
+    case ExtractionBackend.UNSTRUCTURED_OPENSOURCE: {
+      const job = await pythonQueue.add(QueueName.FILES_EXTRACTION_PYTHON, {
+        fileId: file.id,
+        backend
+      });
+      file.extraction = new UnstructuredOpensourceExtraction({ jobId: job.id });
+      await ORM.em.flush();
+      break;
+    }
+    case ExtractionBackend.UNSTRUCTURED_API: {
+      const job = await pythonQueue.add(QueueName.FILES_EXTRACTION_PYTHON, {
+        fileId: file.id,
+        backend
+      });
+      file.extraction = new UnstructuredAPIExtraction({ jobId: job.id });
+      await ORM.em.flush();
+      break;
+    }
+  }
+}
+
+export async function removeExtraction(file: Loaded<File>) {
+  const extraction = file.extraction;
+  if (!extraction) throw new Error('No extraction to remove');
+  switch (extraction.backend) {
+    case ExtractionBackend.WDU:
+      if (extraction.storageId)
+        await s3Client
+          .deleteObject({ Bucket: S3_BUCKET_FILE_STORAGE, Key: extraction.storageId })
+          .promise();
+      break;
+    case ExtractionBackend.UNSTRUCTURED_OPENSOURCE:
+    case ExtractionBackend.UNSTRUCTURED_API:
+      if (extraction.storageId)
+        await s3Client
+          .deleteObject({ Bucket: S3_BUCKET_FILE_STORAGE, Key: extraction.storageId })
+          .promise();
+      break;
+  }
+  file.extraction = undefined;
+  await ORM.em.flush();
+}
+
+export async function getExtractedText(file: Loaded<File>) {
+  const extraction = file.extraction;
+  if (!extraction) throw new Error('Extraction not found');
+  switch (extraction.backend) {
+    case ExtractionBackend.WDU: {
+      if (!extraction.storageId) throw new Error('Extraction missing');
+      const object = await s3Client
+        .getObject({
+          Bucket: S3_BUCKET_FILE_STORAGE,
+          Key: extraction.storageId
+        })
+        .promise();
+      const body = object.Body;
+      if (!body) throw new Error('Invalid Body of a file');
+      return body.toString('utf-8');
+    }
+    case ExtractionBackend.UNSTRUCTURED_OPENSOURCE:
+    case ExtractionBackend.UNSTRUCTURED_API: {
+      if (!extraction.storageId) throw new Error('Extraction missing');
+      const object = await s3Client
+        .getObject({
+          Bucket: S3_BUCKET_FILE_STORAGE,
+          Key: extraction.storageId
+        })
+        .promise();
+      const body = object.Body;
+      if (!body) throw new Error('Invalid Body of a file');
+      const data = body.toString('utf-8');
+      const elements = JSON.parse(data) as UnstructuredExtractionDocument;
+      return elements.map((element) => element.text).join('');
+    }
+  }
+}
+
+export async function getExtractedChunks(file: Loaded<File>) {
+  const extraction = file.extraction;
+  if (!extraction) throw new Error('Extraction not found');
+  switch (extraction.backend) {
+    case ExtractionBackend.WDU: {
+      const text = await getExtractedText(file);
+      const splitter = new RecursiveCharacterTextSplitter();
```

Comment:
There should be some max length otherwise we will exceed the maximum tokens for the embedding model.

This code was important:
```
export const VECTOR_STORE_DEFAULT_MAX_CHUNK_SIZE_TOKENS = 400;
export const VECTOR_STORE_DEFAULT_CHUNK_OVERLAP_TOKENS = 200;

let chunkOverlap = chunkingStrategy.chunk_overlap_tokens;
let chunkSize = chunkingStrategy.max_chunk_size_tokens;

// Todo Tokenize
chunkOverlap *= CHARS_PER_TOKEN_AVG;
chunkSize *= CHARS_PER_TOKEN_AVG;

const splitter = new RecursiveCharacterTextSplitter({ chunkSize, chunkOverlap });
```

Reply:
👍 thanks, missed that